proxy/ptunnel/tunnel.go

```diff
@@ -556,16 +558,17 @@ func (t *Tunnel) BrowserReceive() error {
 			continue
 		}
 
-		size, err := r.Read(ackbuffer[:])
-		if err != nil {
-			t.browser.Error(conn, fmt.Errorf("browser ack read failed with %w", err))
-			continue
+		for ackread := 0; ackread < len(ackbuffer); {
```

Comment:
Could use comment with intent ("retry read until ackbuffer is full" or similar)

Reply:
done.